xds/internal/xdsclient/authority.go

```diff
@@ -548,7 +564,7 @@ func (a *authority) handleWatchTimerExpiryLocked(rType xdsresource.Type, resourc
 	state.md = xdsresource.UpdateMetadata{Status: xdsresource.ServiceStatusNotExist}
 	for watcher := range state.watchers {
 		watcher := watcher
-		a.serializer.TrySchedule(func(context.Context) { watcher.OnResourceDoesNotExist() })
+		a.serializer.TrySchedule(func(context.Context) { watcher.OnResourceDoesNotExist(xdsresource.NopDoneNotifier{}) })
```

Comment:
Same here, why a no-op?

Reply:
Same here. The xDS client's transport is not waiting for any update to be processed. This is a callback that the xDS client authority invokes on its own because of the timer expiry.